xbmc/guilib/guiinfo/GUIInfoLabels.h

```diff
@@ -173,27 +174,31 @@
 #define MUSICPLAYER_RATING          219
 #define MUSICPLAYER_COMMENT         220
 #define MUSICPLAYER_LYRICS          221
-#define MUSICPLAYER_HASPREVIOUS     222
-#define MUSICPLAYER_HASNEXT         223
-#define MUSICPLAYER_EXISTS          224
-#define MUSICPLAYER_PLAYLISTPLAYING 225
-#define MUSICPLAYER_ALBUM_ARTIST    226
-#define MUSICPLAYER_PLAYCOUNT       227
-#define MUSICPLAYER_LASTPLAYED      228
-#define MUSICPLAYER_CHANNEL_NAME    229
-// unused 230
-#define MUSICPLAYER_CHANNEL_GROUP   231
-// unused 232
-#define MUSICPLAYER_CHANNEL_NUMBER  233
-#define MUSICPLAYER_CONTENT         234
-#define MUSICPLAYER_USER_RATING     235
-#define MUSICPLAYER_RATING_AND_VOTES     236
-#define MUSICPLAYER_VOTES           237
-#define MUSICPLAYER_MOOD            238
-#define MUSICPLAYER_CONTRIBUTORS    239
-#define MUSICPLAYER_CONTRIBUTOR_AND_ROLE 240
-#define MUSICPLAYER_DBID            241
-#define MUSICPLAYER_PROPERTY        242
+#define MUSICPLAYER_ALBUM_ARTIST    222
+#define MUSICPLAYER_PLAYCOUNT       223
+#define MUSICPLAYER_LASTPLAYED      224
+#define MUSICPLAYER_USER_RATING     225
+#define MUSICPLAYER_RATING_AND_VOTES     226
+#define MUSICPLAYER_VOTES           227
+#define MUSICPLAYER_MOOD            228
+#define MUSICPLAYER_CONTRIBUTORS    229
+#define MUSICPLAYER_CONTRIBUTOR_AND_ROLE 230
+#define MUSICPLAYER_DBID            231
+
+// Range of musicplayer infolabels that work with offset and position
+#define MUSICPLAYER_OFFSET_POSITION_FIRST MUSICPLAYER_TITLE
+#define MUSICPLAYER_OFFSET_POSITION_LAST MUSICPLAYER_DBID 
```

Comment:
May I suggest to add MUSICPLAYER_OFFSET_POSITION_FIRST directly before the line that defines MUSICPLAYER_TITLE and 
MUSICPLAYER_OFFSET_POSITION_LAST directly after the line that defines MUSICPLAYER_DBID?

Makes it easier to not overlook that the FIRST and LAST defines must be adjusted in case a new first label is added (not very likely, though) and/or a new last label is added (seems to be likely).

Reply:
Somehow I felt I couldn't ref a contant above where it was defined i.e. MUSICPLAYER_TITLE has to come above MUSICPLAYER_OFFSET_POSITION_FIRST, hence put then together at the bottom

Reply:
I thought of something like this:

```c++
#define MUSICPLAYER_OFFSET_POSITION_FIRST MUSICPLAYER_TITLE
#define MUSICPLAYER_TITLE ...
...
#define MUSICPLAYER_DBID ...
#define MUSICPLAYER_OFFSET_POSITION_LAST MUSICPLAYER_DBID
```

This way the both occurences of current first and last label are located as close together as possible, which reduces the risk that LAST and/or FIRST are forgotten to be changed when a new first/last label gets added.

Reply:
Yes I understood what you were thinking, but I thought that the order in which macros were defined mattered, so that a macro had to be defined before it was used in another macro. If what you suggest works on all platforms then sure.

Reply:
Haha, ofc! START needs a bit of tweaking.

Like here https://github.com/xbmc/xbmc/blob/master/xbmc/guilib/guiinfo/GUIInfoLabels.h#L512 ff.

Reply:
Yes, but something for an improvement PR I think. 
Is what is here acceptable as an intermediate step, not too disruptive so can backport as fix?

Reply:
> Is what is here acceptable as an intermediate step

 imo, yes.